src/librustc_driver/driver.rs

```diff
@@ -934,6 +934,7 @@ pub fn phase_3_run_analysis_passes<'tcx, F, R>(sess: &'tcx Session,
     passes.push_pass(MIR_CONST, mir::transform::rustc_peek::SanityCheck);
 
     // What we need to run borrowck etc.
+    passes.push_pass(MIR_CONST, mir::transform::borrow_check::BorrowckMir);
```

Comment:
> So.. why would typeck be a pass while borrowck is a query?

MIR typeck ATM is basically a hacky sanity check. I imagine MIR borrowck/regionck will subsume it.